### PR TITLE
Improve inbox

### PR DESCRIPTION
**Pull request by ariunzayarin**
Add ErrorBoundary function
Fix Send an email modal
Check with develop branch

### CODE REVIEW

File: packages/erxes-ui/src/errorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    // You can also log error messages to an error reporting service here
+  }
+
+  render() {
+    if (this.state.errorInfo) {
+      // Error path
+      Alert.error(this.state.error.toString(), 10000);
+      return (
+          <div>
+          <h5>Error occured.</h5>
```
Comment (Anu-Ujin) [inline packages/erxes-ui/src/errorBoundary.tsx:25]: wrap with translation __

File: packages/erxes-ui/src/errorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          <details style={{ whiteSpace: 'pre-wrap' }}>
```
Comment (Anu-Ujin) [inline packages/erxes-ui/src/errorBoundary.tsx:26]: fix this with styled component

File: packages/plugin-inbox-ui/.erxes/plugin-src/forms/components/ResponseRow.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// import dayjs from 'dayjs';
+// import FormControl from '@erxes/ui/src/components/form/Control';
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/.erxes/plugin-src/forms/components/ResponseRow.tsx:2]: undo comment views and fix

File: packages/plugin-inbox-ui/.erxes/plugin-src/inbox/containers/conversationDetail/ConversationDetail.tsx [hunk 4]
```diff
@@ -130,7 +131,7 @@ class DetailContainer extends React.Component<FinalProps> {
       loading
     };
 
-    return <ConversationDetail {...updatedProps} />;
+    return <ErrorBoundary><ConversationDetail {...updatedProps} /></ErrorBoundary>;
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/.erxes/plugin-src/inbox/containers/conversationDetail/ConversationDetail.tsx:134]: remove wrap with <ErrorBoundary

File: packages/plugin-inbox-ui/.erxes/plugin-src/inbox/containers/leftSidebar/Sidebar.tsx [hunk 5]
```diff
@@ -95,7 +96,7 @@ class Sidebar extends React.Component<FinalProps> {
       );
     };
 
-    return <Bulk content={content} />;
+    return <ErrorBoundary><Bulk content={content} /></ErrorBoundary>;
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/.erxes/plugin-src/inbox/containers/leftSidebar/Sidebar.tsx:99]: remove this too

File: packages/plugin-inbox-ui/src/forms/components/ResponseRow.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+// import dayjs from 'dayjs';
+// import FormControl from '@erxes/ui/src/components/form/Control';
+// import Icon from '@erxes/ui/src/components/Icon';
+// import Label from '@erxes/ui/src/components/Label';
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/src/forms/components/ResponseRow.tsx:4]: undo comments and fix

File: packages/plugin-inbox-ui/src/forms/containers/ResponseList.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  fieldsQuery: FieldsQueryResponse;
+  formSubmissionTotalCountQuery: FormSubmissionsTotalCountQueryResponse;
+} & IRouterProps &
+  Props;
+
+class ListContainer extends React.Component<FinalProps> {
+  // componentDidMount() {
+  //   const { history } = this.props;
+
+  //   const shouldRefetchList = routerUtils.getParam(history, 'popUpRefetchList');
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/src/forms/containers/ResponseList.tsx:36]: undo comments and fix

File: packages/plugin-inbox-ui/src/inbox/containers/conversationDetail/ConversationDetail.tsx [hunk 8]
```diff
@@ -130,7 +131,7 @@ class DetailContainer extends React.Component<FinalProps> {
       loading
     };
 
-    return <ConversationDetail {...updatedProps} />;
+    return <ErrorBoundary><ConversationDetail {...updatedProps} /></ErrorBoundary>;
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/src/inbox/containers/conversationDetail/ConversationDetail.tsx:134]: remove wrong wrap with error boundary

File: packages/plugin-inbox-ui/src/inbox/containers/leftSidebar/Sidebar.tsx [hunk 9]
```diff
@@ -95,7 +96,7 @@ class Sidebar extends React.Component<FinalProps> {
       );
     };
 
-    return <Bulk content={content} />;
+    return <ErrorBoundary><Bulk content={content} /></ErrorBoundary>;
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/src/inbox/containers/leftSidebar/Sidebar.tsx:99]: remove

File: packages/plugin-inbox-ui/src/settings/integrations/containers/messenger/AddOns.tsx [hunk 10]
```diff
@@ -42,7 +42,7 @@ class KnowledgeBaseContainer extends React.Component<FinalProps> {
 
     const updatedProps = {
       ...this.props,
-      topics,
+      topics: topics as any,
```
Comment (Anu-Ujin) [inline packages/plugin-inbox-ui/src/settings/integrations/containers/messenger/AddOns.tsx:45]: change with ITopic[] type

File: packages/ui-inbox/src/inbox/styles.ts [hunk 11]
```diff
@@ -345,6 +345,20 @@ const CallLabel = styledTS<{ type: string }>(styled.span)`
   color: ${props => (props.type === 'answered' ? 'green' : 'red')};
 `;
 
+// ui-engage ruu oruulah
```
Comment (Anu-Ujin) [inline packages/ui-inbox/src/inbox/styles.ts:348]: remove comment or check and fix

File: packages/ui-inbox/src/inbox/utils.ts [hunk 12]
```diff
@@ -87,3 +87,12 @@ export const urlify = (text: string) => {
 };
 
 export const hasAnyActivity = activityLogsUtils.hasAnyActivity;
+
+// ui-engage ruu oruulah
```
Comment (Anu-Ujin) [inline packages/ui-inbox/src/inbox/utils.ts:91]: fix or remove

File: packages/ui-leads/package.json [hunk 13]
```diff
@@ -7,6 +7,7 @@
     "react": "16.14.0",
     "react-apollo": "3.1.3",
     "react-router-dom": "5.1.2",
+    "react-select-plus": "^1.0.0-rc.5",
```
Comment (Anu-Ujin) [inline packages/ui-leads/package.json:10]: is this the right version? double check

File: packages/ui-inbox/src/inbox/styles.ts (hunk 11 above)
Comment (ariunzayarin) [reply packages/ui-inbox/src/inbox/styles.ts:348]: Enkhzul appollo-federation-ui bolon ui-test 2iig merge hiisnii daraa shalgaad ui-engage deer bvl ustgaarai gsn tul turdee oruulsan bgaa. Odoo shuud ui-engage ruu ni oruulah uu esvel merge hiisnii daraa boloh uu?

**Pull request merged**